src/propeller/push/instructions/polymorphic.cljc

```diff
@@ -38,7 +44,8 @@
             (and (not= stack :integer)
                  (not (state/empty-stack? state :integer))
                  (not (state/empty-stack? state stack))))
-      (let [n (state/peek-stack state :integer)
+      (let [n (min (state/peek-stack state :integer)
+                   (inc (- max-stack-items (state/stack-size state stack))))
```

Comment:
You need the `inc` (which corresponds to the `(- n 1)` on line 51) because you're removing _one_ copy of `x` and then pushing `n` copies back on, so you're actually only adding `(- n 1)` _new_ copies. This took us a while to wrap our heads around. The comment before `_dup_times` was quite helpful 👍 , but maybe something about this should be documented here as well?